stackit/internal/services/logme/instance/resource.go

```diff
@@ -299,6 +301,11 @@ func (r *instanceResource) Read(ctx context.Context, req resource.ReadRequest, r
 
 	instanceResp, err := r.client.GetInstance(ctx, projectId, instanceId).Execute()
 	if err != nil {
+		oapiErr, ok := err.(*oapierror.GenericOpenAPIError) //nolint:errorlint //complaining that error.As should be used to catch wrapped errors, but this error should not be wrapped
+		if ok && (oapiErr.StatusCode == http.StatusNotFound) || (oapiErr.StatusCode == http.StatusGone) {
+			resp.State.RemoveResource(ctx)
```

Comment:
if the resource was never in the state, this function doesn't do anything and doesn't throw error right?

Reply:
Yes